### PR TITLE
Feature/gpsservice

### PR DESCRIPTION
**Pull request by matteopariset**
GPS location acquisition, analysis and storage is done by the background service LocationService

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/DataForDemo.java [hunk 1]
```diff
@@ -11,6 +14,7 @@
 import ch.epfl.sdp.contamination.DataSender;
 import ch.epfl.sdp.contamination.GridFirestoreInteractor;
 import ch.epfl.sdp.contamination.Layman;
+import kotlin.NotImplementedError;
```
Comment (Ulydev) [inline app/src/androidTest/java/ch/epfl/sdp/DataForDemo.java]: could also use UnsupportedOperationException, standard to java, but that does the job too :-)

**Review by Ulydev — APPROVED** [review]
great code! and permissions are well handled 👍

File: app/src/androidTest/java/ch/epfl/sdp/DataForDemo.java (hunk 1 above)
Comment (matteopariset) [reply app/src/androidTest/java/ch/epfl/sdp/DataForDemo.java]: You're right, will use that one

File: app/src/androidTest/java/ch/epfl/sdp/GpsActivityTest.java [hunk 2]
```diff
@@ -45,9 +53,26 @@
     @Rule
     public ExpectedException illegalArgument = ExpectedException.none();
 
-    private void startActivityWithBroker(LocationBroker br) {
+    private void startActivityWithBroker(LocationBroker br) throws Throwable {
```
Comment (AdrienRem) [inline app/src/androidTest/java/ch/epfl/sdp/GpsActivityTest.java:56]: Can't you use a more specific exception than Throwable ?

Comment (matteopariset) [reply app/src/androidTest/java/ch/epfl/sdp/GpsActivityTest.java:56]: Nope. Throwable is the most specific object thrown by runOnUiThread()

Comment (AdrienRem) [reply app/src/androidTest/java/ch/epfl/sdp/GpsActivityTest.java:56]: okay !

**Review by AdrienRem — APPROVED** [review]

**Pull request merged**